video.go

```diff
@@ -53,6 +54,52 @@ func (v *Video) parseVideoInfo(info string) error {
 		v.Duration = time.Duration(seconds) * time.Second
 	}
 
+	// Check if video is downloadable
```

Comment:
duplicate code like line 103-121,  please try to extract a method.

Reply:
Merged the 2 methods into a single method `parseVideoInfoOrPage` that has an extra `bool` parameter `isVideoPage`.